projects/noloan2/app/src/main/java/hasadna/noloan2/LawsuitActivity.java

```diff
@@ -184,121 +186,64 @@ private String fillTemplate() throws IOException {
+    lawsuit = lawsuit.replace("<receivedSpamDate>", selectedSmsSpam.getReceivedAt());
     lawsuit = lawsuit.replace("<currentDate>", DATE_FORMATTER.format(new Date()));
-
-    // User fields
-    lawsuit =
-        lawsuit.replace(
-            "<userFullName>",
-            ((EditText) findViewById(R.id.userPrivateName)).getText().toString()
-                + " "
-                + ((EditText) findViewById(R.id.userLastName)).getText().toString());
-    lawsuit =
-        lawsuit.replace("<userId>", ((EditText) findViewById(R.id.userID)).getText().toString());
-    lawsuit =
-        lawsuit.replace(
-            "<userAddress>", ((EditText) findViewById(R.id.userAddress)).getText().toString());
-    lawsuit =
-        lawsuit.replace(
-            "<userPhone>", ((EditText) findViewById(R.id.userPhone)).getText().toString());
-    lawsuit =
-        lawsuit.replace("<userFax>", ((EditText) findViewById(R.id.userFax)).getText().toString());
-
-    // Spam company
-    lawsuit =
-        lawsuit.replace(
-            "<companyName>", ((EditText) findViewById(R.id.companyName)).getText().toString());
-    lawsuit =
-        lawsuit.replace(
-            "<companyId>", ((EditText) findViewById(R.id.companyId)).getText().toString());
-    lawsuit =
-        lawsuit.replace(
-            "<companyAddress>",
-            ((EditText) findViewById(R.id.companyAddress)).getText().toString());
-    lawsuit =
-        lawsuit.replace(
-            "<companyPhone>", ((EditText) findViewById(R.id.companyPhone)).getText().toString());
-    lawsuit =
-        lawsuit.replace(
-            "<companyFax>", ((EditText) findViewById(R.id.companyFax)).getText().toString());
-
+    lawsuit = lawsuit.replace("<undefined>", "");
     return lawsuit;
   }
 
-  // Pops a dialog asking the user to share the pdf
-  private void sharePdf(String absFilename) {
-    if (new File(absFilename).exists()) {
-      new AlertDialog.Builder(this, R.style.AlertDialog)
-          .setTitle("כתב תביעה נוצר")
-          .setMessage("כתב התביעה נוצר ונשמר במכשירך.\n האם תרצה/י לשתף?")
-          .setPositiveButton(
-              "כן",
-              (dialog, which) -> {
-                Uri uri =
-                    FileProvider.getUriForFile(
-                        getApplicationContext(),
-                        getPackageName() + ".fileprovider",
-                        new File(absFilename));
-                Intent shareIntent = new Intent();
-                shareIntent.setAction(Intent.ACTION_SEND);
-                shareIntent.setType("application/pdf");
-                shareIntent.putExtra(Intent.EXTRA_STREAM, uri);
-                startActivity(Intent.createChooser(shareIntent, "Share via"));
-              })
-          .setNegativeButton("לא", (dialog, which) -> dialog.dismiss())
-          .create()
-          .show();
+  // Create a folder for generated PDFs (external storage)
+  private void createOutputDir() {
```

Comment:
Helper functions suppose to be below the function it help not above

Reply:
:+1: